resources/Microsoft.Windows.Settings/Microsoft.Windows.Settings.psm1

```diff
@@ -49,34 +74,56 @@ class WindowsSettings {
 
+        # Get OS Version
+        $currentState.OsVersion = $this.GetOsVersion()
+
+        # Get File Extensions visibility
+        $currentState.HideFileExt = $this.IsFileExtensionsHidden()
+
+        # Get Hidden Files visibility
+        $currentState.ShowHiddenFiles = $this.AreHiddenFilesShown()
+
+        # Get Long Paths Enabled
+        $currentState.LongPathsEnabled = $this.IsLongPathsEnabled()
+
         return $currentState
     }
 
     [bool] Test() {
         $currentState = $this.Get()
-        return $this.TestTaskbarAlignment($currentState) -and $this.TestAppColorMode($currentState) -and $this.TestSystemColorMode($currentState) -and $this.TestDeveloperMode($currentState)
+
+        return (
+            $this.TestTaskbarAlignment($currentState) -and
+            $this.TestAppColorMode($currentState) -and
+            $this.TestSystemColorMode($currentState) -and
+            $this.TestDeveloperMode($currentState) -and
+            $this.TestOsVersion($currentState) -and
+            $this.TestHideFileExt($currentState) -and
+            $this.TestHiddenFilesShown($currentState) -and
+            $this.TestLongPathsEnabled($currentState)
+        )
     }
 
     [void] Set() {
         $currentState = $this.Get()
 
         # Set TaskbarAlignment
-        if (!$this.TestTaskbarAlignment($currentState)) {
-            $desiredAlignment = $this.TaskbarAlignment -eq "Left" ? 0 : 1
-            Set-ItemProperty -Path $global:ExplorerRegistryPath -Name $this.TaskbarAl -Value $desiredAlignment
+        if (!$this.TestTaskbarAlignment($currentState) -and @($this.LeftValue, $this.CenterValue) -contains $this.TaskbarAlignment) {
+            $desiredAlignment = $this.TaskbarAlignment -eq $this.LeftValue ? 0 : 1
+            Set-ItemProperty -Path $global:ExplorerRegistryPath -Name $this.TaskbarAlignmentRegKey -Value $desiredAlignment
         }
 
         # Set ColorMode
         $colorModeChanged = $false
-        if (!$this.TestAppColorMode($currentState)) {
-            $desiredColorMode = $this.AppColorMode -eq "Dark" ? 0 : 1
-            Set-ItemProperty -Path $global:PersonalizeRegistryPath -Name $this.AppsUseLightTheme -Value $desiredColorMode
+        if (!$this.TestAppColorMode($currentState) -and @($this.DarkValue, $this.LightValue) -contains $this.AppColorMode) {
+            $desiredColorMode = $this.AppColorMode -eq $this.DarkValue ? 0 : 1
+            Set-ItemProperty -Path $global:PersonalizeRegistryPath -Name $this.AppsUseLightThemeRegKey -Value $desiredColorMode
             $colorModeChanged = $true
         }
 
-        if (!$this.TestSystemColorMode($currentState)) {
-            $desiredColorMode = $this.SystemColorMode -eq "Dark" ? 0 : 1
-            Set-ItemProperty -Path $global:PersonalizeRegistryPath -Name $this.SystemUsesLightTheme -Value $desiredColorMode
+        if (!$this.TestSystemColorMode($currentState) -and @($this.DarkValue, $this.LightValue) -contains $this.SystemColorMode) {

```

Comment:
While I don't know what the DSC way/convention is to handle invalid inputs, currently the `Set` method will do nothing if met with a bad input (example setting SystemColorMode to anything other than "Light" / "Dark"). 

If we take a look at the GitDsc resource, there are numerous places where we throw a custom error message 

https://github.com/microsoft/winget-dsc/blob/b752a111dfc5ea8b48f721bc34f62045efb74155/resources/GitDsc/GitDsc.psm1#L109-L111


Should we throw custom error messages here as well?

Reply:
Good idea! I added an exception which would validate and print the expected values ☺️